.github/workflows/main.yml

```diff
@@ -0,0 +1,72 @@
+name: CI
+
+on:
+  push:
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: windows-latest  # Use Ubuntu in v5.0
+
+    env:
+      DOTNET_NOLOGO: true
+      DOTNET_SKIP_FIRST_TIME_EXPERIENCE: true # Avoid pre-populating the NuGet package cache
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # all
+
+      - name: Setup .NET 2.0 # Remove in v5.0
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 2.0.x
+
+      - name: Setup .NET 6.0
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 6.0.x
+
+      - name: Install GitVersion
+        uses: gittools/actions/gitversion/setup@v0.9.7
+        with:
+          versionSpec: '5.x'
+
+      - name: Run GitVersion
+        uses: gittools/actions/gitversion/execute@v0.9.7
+
+      - name: Display SemVer
+        run: |
+            echo "SemVer: $env:GitVersion_SemVer"
+          
+      - name: Add DbUp NuGet Source
+        run: dotnet nuget add source --name DbUp --username DbUp --password ${{ secrets.GITHUB_TOKEN }} --store-password-in-clear-text https://nuget.pkg.github.com/DbUp/index.json
+        
+      - name: Restore
+        run: dotnet restore
+        working-directory: src
+
+      - name: Build
+        run: dotnet build -c Release --no-restore /p:Version=$env:GitVersion_SemVer
+        working-directory: src
+
+      - name: Test
+        run: dotnet test --no-build -c Release --logger trx --logger "console;verbosity=detailed" --results-directory ../artifacts
+        working-directory: src
+
+      - name: Pack
+        run: dotnet pack --no-build -c Release -o ../artifacts /p:Version=$env:GitVersion_SemVer
+        working-directory: src
+
+      - name: Push NuGet packages to GitHub Packages ⬆️
+        working-directory: artifacts
+        run: dotnet nuget push dbup-redshift.$env:GitVersion_SemVer.nupkg --api-key ${{ secrets.GITHUB_TOKEN }} --source "https://nuget.pkg.github.com/DbUp/index.json"
```

Comment:
Just wondering if we need every PR & manual build to push to GitHub Packages? Every main line build, sure. But if you need to test a PR's build you can download the artifacts & reference from a folder, right?

Reply:
The build doesn't currently attach artifacts (anymore). For the `dbup-core` project it'd be good to keep the push so the providers could use a pre-release/branch and we can co-ordinate.

The only reason I can think not to is space. It looks like we have 500mb to play with, approx 10,000 packages.